### PR TITLE
Refactor pull request related methods into a more centralized location

### PR DESCRIPTION
**Pull request by codenamev**
Refactor pull request related methods into a more centralized location

### CODE REVIEW

File: lib/git_reflow/git_server/bit_bucket/pull_request.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        def last_comment
+          last_comment = comments.first
+          return "" unless last_comment
+          "#{last_comment.content.raw}"
+        end
+
+        def reviewers
+          return [] unless comments.size > 0
+          comments.map {|c| c.user.username } - [GitReflow.git_server.class.user]
```
Comment (nhance) [inline lib/git_reflow/git_server/bit_bucket/pull_request.rb]: Does this fix the issue where we would require a lgtm from ourself?

File: lib/git_reflow/git_server/git_hub/pull_request.rb [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            nil
+          end
+        end
+
+        def reviewers
+          comment_authors
+        end
+
+        def approvals
+          pull_last_committed_at = get_commited_time(self.head.sha)
```
Comment (nhance) [inline lib/git_reflow/git_server/git_hub/pull_request.rb]: This is picky, but you have `committed` (correct) and `commited` (wrong spelling) used here

File: lib/git_reflow/git_server/pull_request.rb [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+      def approvals
+        raise "#{self.class.to_s}#approvals method must be implemented"
+      end
+
+      def reviewers_pending_response
+        reviewers - approvals
+      end
+
+      def good_to_merge?(force: false)
```
Comment (nhance) [inline lib/git_reflow/git_server/pull_request.rb]: I like how you've abstracted this so it isn't implemented elsewhere. Very clean!

File: spec/lib/git_reflow/config_spec.rb [hunk 4]
```diff
@@ -6,8 +6,24 @@
     it      { expect{ subject }.to have_run_command_silently 'git config --get chucknorris.roundhouse' }
 
     context "and getting all values" do
-      subject { GitReflow::Config.get('chucknorris.roundhouse', all: true) }
-      it      { expect{ subject }.to have_run_command_silently 'git config --get-all chucknorris.roundhouse' }
+      subject { GitReflow::Config.get('chucknorris.roundhouse-kick', all: true) }
+      it      { expect{ subject }.to have_run_command_silently 'git config --get-all chucknorris.roundhouse-kick' }
+
+      context "and checking locally only" do
```
Comment (nhance) [inline spec/lib/git_reflow/config_spec.rb]: When chucknorris jumps, does he jump off the planet, or does the planet move away from chucknorris?

File: spec/lib/git_server/bit_bucket/pull_request_spec.rb [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    allow(bitbucket.class).to receive(:api_endpoint).and_return(api_endpoint)
+    allow(bitbucket.class).to receive(:remote_user).and_return(user)
+    allow(bitbucket.class).to receive(:remote_repo).and_return(repo)
+    allow(bitbucket.class).to receive(:oauth_token).and_return(oauth_token_hash.token)
+    allow(bitbucket.class).to receive(:get_commited_time).and_return(Time.now)
+
+    allow(GitReflow).to receive(:git_server).and_return(bitbucket)
+  end
+
+  xdescribe '.create(options)' do
```
Comment (nhance) [inline spec/lib/git_server/bit_bucket/pull_request_spec.rb]: Do we want to leave the pending tests in here, or are they just dead weight?

Comment (nhance) [inline spec/lib/git_server/bit_bucket/pull_request_spec.rb]: It feels like dead weight unless we actually have tests written.

File: spec/support/fake_github.rb [hunk 6]
```diff
@@ -0,0 +1,128 @@
+require 'rspec/mocks'
+require 'webmock'
+require 'chronic'
+
+class FakeGitHub
```
Comment (nhance) [inline spec/support/fake_github.rb]: This is cool

File: lib/git_reflow/git_server/bit_bucket/pull_request.rb (hunk 1 above)
Comment (codenamev) [inline lib/git_reflow/git_server/bit_bucket/pull_request.rb]: it should, I wasn't able to get a failure working with that scenario

File: lib/git_reflow/git_server/git_hub/pull_request.rb (hunk 2 above)
Comment (codenamev) [inline lib/git_reflow/git_server/git_hub/pull_request.rb]: not picky, heh, I'm a terrible speller, I'll update

**Comment by nhance** [comment]
lgtm, nice work Val!

**Pull request closed without merging**